engine/time/src/main/java/io/deephaven/time/calendar/BusinessPeriod.java

```diff
@@ -5,56 +5,70 @@
 
 import io.deephaven.time.DateTimeUtils;
 
-import java.io.Serializable;
 import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.Temporal;
+import java.util.Objects;
 
 /**
  * A period of business time during a business day.
+ *
+ * @param <T> time type
  */
-public class BusinessPeriod implements Serializable {
-    private static final long serialVersionUID = 8196837269495115196L;
-    private final Instant startTime;
-    private final Instant endTime;
+public class BusinessPeriod<T extends Comparable<T> & Temporal> {
+    private final T start;
+    private final T end;
+    private final long nanos;
 
-    @SuppressWarnings("ConstantConditions")
-    BusinessPeriod(final Instant startTime, final Instant endTime) {
-        this.startTime = startTime;
-        this.endTime = endTime;
+    /**
+     * Create a new business period.
+     *
+     * @param startTime start of the business period.
+     * @param endTime end of the business period.
+     */
+    BusinessPeriod(final T startTime, final T endTime) {
+        this.start = startTime;
+        this.end = endTime;
 
         if (startTime == null || endTime == null) {
             throw new IllegalArgumentException("Null argument: startTime=" + startTime + " endTime=" + endTime);
         }
 
-        if (DateTimeUtils.epochNanos(startTime) > DateTimeUtils.epochNanos(endTime)) {
+        if (startTime.compareTo(endTime) > 0) {
             throw new IllegalArgumentException("Start is after end: startTime=" + startTime + " endTime=" + endTime);
         }
+
+        this.nanos = start.until(end, ChronoUnit.NANOS);
     }
 
     /**
-     * Returns the start of the period.
+     * Start of the period.
```

Comment:
Can you comment on the type here? "Start of the period as an ISO-formatted..." or "... in nanoseconds since the epoch". Does it depend on how this class is instantiated, since it's templated with `T`?

Reply:
It depends on how it is instantiated.  The comment indicates this:
```java
@param <T> time type
```